options.go

```diff
@@ -0,0 +1,100 @@
+package tflog
+
+import (
+	"io"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+// Option defines a modification to the configuration for a logger.
+type Option func(loggerOpts) loggerOpts
+
+type loggerOpts struct {
+	name            string
+	level           hclog.Level
+	includeLocation bool
+
+	// some private options to be used only by tflog for testing purposes
+	// we should never export an Option that sets these
+	output      io.Writer
+	includeTime bool
+}
+
+func applyLoggerOpts(opts ...Option) loggerOpts {
+	// set some defaults
+	l := loggerOpts{
+		includeLocation: true,
+		includeTime:     true,
+		output:          os.Stderr,
+	}
+	for _, opt := range opts {
+		l = opt(l)
+	}
+	return l
+}
+
+func withOutput(output io.Writer) Option {
+	return func(l loggerOpts) loggerOpts {
+		l.output = output
+		return l
+	}
+}
+
+func withoutTimestamp() Option {
+	return func(l loggerOpts) loggerOpts {
+		l.includeTime = false
+		return l
+	}
+}
+
+// WithLogName returns an option that will set the logger name explicitly to
+// `name`. This has no effect when used with NewSubsystem.
+func WithLogName(name string) Option {
+	return func(l loggerOpts) loggerOpts {
+		l.name = name
+		return l
+	}
+}
+
+// WithLevelFromEnv returns an option that will set the level of the logger
+// based on the string in an environment variable. The environment variable
+// checked will be `name` and `subsystems`, joined by _ and in all caps.
```

Comment:
Perhaps add an example

Reply:
Good call, I'll follow up with another PR for that and some README/doc.go tweaks as part of release prep.